quic/s2n-quic-platform/src/syscall.rs

```diff
@@ -8,13 +8,25 @@ use core::ops::ControlFlow;
 use socket2::{Domain, Protocol, Socket, Type};
 use std::io;
 
+/// Calls the given libc function and wraps the result in an `io::Result`.
+macro_rules! libc {
```

Comment:
doesnt have to be in this PR but all `setsockopt` calls can use a comment explaining what its doing. Otherwise reader has to lookup the call and or be familiar with it.

Reply:
https://github.com/aws/s2n-quic/issues/1815